File: typescript/recurring-products.ts
declare const process: { env: Record<string, string | undefined> };

const baseUrl = process.env.ANTOPS_DIRECT_BASE_URL ?? "https://api.antops.dev";
const apiKey = process.env.ANTOPS_API_KEY;

if (!apiKey) throw new Error("Set ANTOPS_API_KEY before running this example.");

async function call(path: string, options: RequestInit = {}): Promise<unknown> {
  const response = await fetch(`${baseUrl}${path}`, {
    ...options,
    headers: { "content-type": "application/json", "x-api-key": apiKey, ...options.headers },
  });
  if (!response.ok) throw new Error(`${response.status}: ${await response.text()}`);
  return response.json();
}

void (async () => {
  console.log(await call("/v1/companies/GB/00000006"));
  console.log(await call("/v1/tenders"));
  console.log(await call("/v1/change-risk/analyses", {
    method: "POST",
    body: JSON.stringify({ files: [{ path: "Dockerfile", content: "FROM python:3.12-slim" }] }),
  }));
})();

File: python/recurring_products.py
"""Run with ANTOPS_API_KEY=YOUR_WORKSPACE_KEY python recurring_products.py."""

import json
import os
from urllib.request import Request, urlopen

BASE_URL = os.getenv("ANTOPS_DIRECT_BASE_URL", "https://api.antops.dev")
API_KEY = os.environ["ANTOPS_API_KEY"]


def call(method: str, path: str, payload: dict | None = None) -> dict:
    request = Request(
        f"{BASE_URL}{path}",
        data=json.dumps(payload).encode() if payload else None,
        headers={"Content-Type": "application/json", "X-API-Key": API_KEY},
        method=method,
    )
    with urlopen(request, timeout=30) as response:  # nosec B310: fixed HTTPS base URL
        return json.load(response)


print(json.dumps(call("GET", "/v1/companies/GB/00000006"), indent=2))
print(json.dumps(call("GET", "/v1/tenders"), indent=2))
print(
    json.dumps(
        call(
            "POST",
            "/v1/change-risk/analyses",
            {"files": [{"path": "Dockerfile", "content": "FROM python:3.12-slim"}]},
        ),
        indent=2,
    )
)
